rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -0,0 +1,157 @@
+        // It will be false if the extracted inputSigHash is different from the inputSigHash used for generating the signature.
+        Assertions.assertTrue(valid);
+    }
+
+    private static Stream<Arguments> peginArgProvider() {
+        return Stream.of(
+            // PEGIN BECH32 - https://www.blockchain.com/explorer/transactions/btc/aeb98f9a7632efefcd8f9d89b881a0d7a80e4e5c501482f8c9a57db1d7e919c0
+            Arguments.of("010000000001012ee5f3cf0cf707d9fb1233653c3c8dfc96850cdf80ba716d4b4917d1ded876220000000000ffffffff0320a107000000000017a914056d0d9c5b14dd720d9f61fdb3f557c074f95cef8700000000000000001b6a1952534b5401a0df67b9589bd0527af41f66294a846da513e1f9f3c201000000000016001402bd283849dbe761ee7a8e9d902a5dee1f9807f20248304502210080e065d2ed38d819d26869e3ea25918adaa62fbaf1ce3a98e809a5c6a3c0beb10220669234d087835efcbccf21dbf225fee2abea4e37d29abead8a6da660812c450b01210210a19836ab556cc76f66ad8536fb613869db9676d123a7e56a1488369b646ed100000000"),
+            // PEGIN legacy - https://www.blockchain.com/explorer/transactions/btc/529682fd067ba26a8406c809c605cde166d6818d0495f3c32bc74d100422fc30
+            Arguments.of("02000000022ccf58addb0abd17bb89b8371d0359cfe56ea023b0cf0571e0ccd3917889c522000000006a47304402202cc1339e9090b57fa290a2c8506e82b360e07da2deeec0a56789f2623145668b02201b2d5d5c4d67f8ba217de8fc3269f70881b5669b5262daa29130271db7e0f7bd012103dd1adad32f93448dd9c86ea772c24152e9f85be8e116c336079d87f342b97badfdffffffdd9e3403bab1d06816b3547011cffafed2630606a3811b36ce2b61b80e34eab4000000006b483045022100b4986ebe1336d6468ddeec9461374b955e38d55e76f0e4c4f3d78df1c75e4298022052ed4e50f940680c81bc7299b69c81c7122f4250582035669f838328165548c9012103dd1adad32f93448dd9c86ea772c24152e9f85be8e116c336079d87f342b97badfdffffff0225a107000000000017a9142c1bab6ea51fdaf85c8366bd2b1502eaa69b6ae6871f221800000000001976a9140eda35d81e2a8537beebbbcdf63e3483be01269288acb1cd0b00"),
+            // FLYOVER PEGIN - https://live.blockcypher.com/btc-testnet/tx/d3356587d46ccc887c577eff8d78bff4e81735fd5fcd6511499edc04d7f0c4e9/
+            Arguments.of("020000000161733fc13d12e74a99ba4f79ea3cbcb6dc1714d21474b0611b2fe5adbfb592ef020000006a473044022016f565d5a69195deb489101ff20af2bca28a38f15915e9ad773b1b9bd9c6013c022041e61ce5d89182478a8ad1602308a7a1cea0b730552ca8ba2a9d3368f9bd8f9201210205cd73583ce9a42bda0bff53f52398e04c703dd8aa5de03dcf780e26e2ba4ca1fdffffff02237a08000000000017a914012c4ac8c51ba78e04a8531d3c5788e7a6af51e4872f98ab01000000001976a91435309691021c3e612408bcbc63220222e3cf746388acf4282500")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("peginArgProvider")
+    void test_getFirstInputSigHash_pegin(String btcTxHex) {
+        // Arrange
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams, Hex.decode(btcTxHex));
+
+        // Act
+        Optional<Sha256Hash> firstInputSigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
+        // Assert
+        Assertions.assertFalse(firstInputSigHash.isPresent());
```

Comment:
Why is there no sig hash?

Reply:
There are three tx being processed for this test:

1 - The first one is a PEGIN BECH32. ScriptSig is empty. No chunks. If no redeemScript is found, no sighash is returned. This is a limitation of `BitcoinUtils#getFirstInputSigHash`. It only works for txs where the first input has a redeem script.
2 - Second, is a PEGIN legacy. It failed to extract the redeem script. Since no redeem script is extracted, no sighash can be generated.
3- Third one, the same as the second.

Do you think we should change the name of the method to be explicit that it works only for bridge pegout transactions?

Reply:
Sounds like three different test scenarios, not sure if it makes sense to have them all in the same method.

First case, makes sense once you explain it but not when reading the test. Maybe it should be part of the test name.

Second case, no direct relationship between being a pegin legacy and not having a redeem script. It doesn't have a redeem script because the sender is a P2PKH address and not a P2SH. It is true that legacy peg-ins can only be done from P2PKH addresses but still, sounds a bit stretched. I'd simply test that it will fail to find the sighash of a P2PKH transaction. Not related to peg-ins. If we want to test peg-ins, why not test peg-in v1 where the sender can be P2SH?

Reply:
Done